src/main/java/io/spokestack/spokestack/tts/TTSCallback.java

```diff
@@ -12,14 +12,16 @@
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
+import java.util.concurrent.ConcurrentLinkedQueue;
 
 /**
  * An interface used to receive playback URLs asynchronously from the Spokestack
  * TTS service.
  */
 public abstract class TTSCallback implements Callback {
     private final Gson gson = new Gson();
-    private List<QueuedCall> callQueue = new ArrayList<>();
+    private final ConcurrentLinkedQueue<QueuedCall> callQueue =
```

Comment:
using a lock-free queue isn't the solution here, because you can still be iterating over the queue while it is being modified, which can cause duplicate response delivery
the simplest solution is to lock the queue in `enqueue` and `onResponse`